client/my-sites/earn/components/add-edit-coupon-modal/test/products-selector.jsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import membershipsReducer from 'calypso/state/memberships/reducer';
+import siteSettingsReducer from 'calypso/state/site-settings/reducer';
+import uiReducer from 'calypso/state/ui/reducer';
+import { renderWithProvider } from '../../../../../test-helpers/testing-library';
+import ProductsSelector from '../products-selector';
+
+const productData = {
+	currency: 'USD',
+	buyer_can_change_amount: false,
+	multiple_per_user: false,
+	welcome_email_content: 'Welcome!',
+	subscribe_as_site_subscriber: true,
+	type: 'newsletter',
+	is_editable: false,
+};
+
+const testProduct1 = {
+	...productData,
+	ID: 9,
+	price: 5,
+	title: 'Monthly Subscription',
+	interval: '1 month',
+	renewal_schedule: '1 month',
+};
+
+const testProduct2 = {
+	...productData,
+	ID: 10,
+	price: 40,
+	title: 'Yearly Subscription',
+	interval: '1 year',
+	renewal_schedule: '1 year',
+};
+
+const initialState = {
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {},
+		},
+		productList: {
+			items: {
+				1: [ testProduct1, testProduct2 ],
+			},
+		},
+	},
+};
+
+describe( 'ProductsSelector', () => {
+	const getAnyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', { name: 'Any product' } );
+	const getYearlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Yearly Subscription : $40.00 / year',
+		} );
+	const getMonthlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Monthly Subscription : $5.00 / month',
+		} );
+	const getDropdownButton = ( name ) => screen.getByRole( 'button', { name } );
+
+	test( 'should render with any product selected when no specific products have been selected', async () => {
+		const render = ( el, options ) =>
+			renderWithProvider( el, {
+				...options,
+				initialState,
+				reducers: {
+					ui: uiReducer,
+					memberships: membershipsReducer,
+					siteSettings: siteSettingsReducer,
+				},
+			} );
```

Comment:
This seems repetitive throughout multple tests, perhaps we could move it out and reuse it?